src/index.ts

```diff
@@ -282,3 +282,59 @@ export const gt = (versionA: string | null, versionB: string | null): boolean =>
 export const lt = (versionA: string | null, versionB: string | null): boolean => {
 	return compare(versionA, versionB) < 0;
 };
+
+/**
+ * @summary Check if a version satisfies a range
+ * @name satisfies
+ * @public
+ * @function
+ *
+ * @description Return true if the version satisfies the range
+ *
+ * @param {string|null} version - The version to evaluate
+ * @param {string} range - A semver range string, see the [node-semver](https://github.com/npm/node-semver#ranges)
+ * docs for details
+ *
+ * @returns {boolean} - True if the version satisfies the range, false otherwise
+ *
+ */
+export const satisfies = (version: string | null, range: string) => {
+	if (!version) {
+		return false;
+	}
+
+	version = normalize(<string>version);
+
+	if (!semver.valid(version)) {
+		return false;
+	}
+
+	return semver.satisfies(version, range);
+};
+
+/**
+ * @summary Return the lowest version in the list that satisfies the range
+ * @name maxSatisfying
+ * @public
+ * @function
+ *
+ * @description Return the highest version in the list that satisfies the range, or null if none of them do.
+ *
+ * @param {Array.<string|null>} versions - An array of versions to evaluate
+ * @param {string} range - A semver range string, see the [node-semver](https://github.com/npm/node-semver#ranges)
+ * docs for details
+ *
+ * @returns {string|null} - The highest matching version string, or null.
+ *
+ */
+export const maxSatisfying = (versions: Array<string | null>, range: string) => {
+	let max: string | null = null;
+
+	versions.forEach((version) => {
+		if (satisfies(version, range) && gt(version, max)) {
```

Comment:
How about `max === null || gt(version, max)` to save some nested calls?

My only heads-up is that the `satisfies` in here will run `normalize` & `semver.valid` N times.
How about normalizing & checking for validity outside of the loop and then use the `semver.satisfies`.

Getting with it even further, someone might think that `maxSatisfying` can be the base plural/composite implementation of `satisfies`.
In other words:
```js
satisfies(version, range) = maxSatisfying([version], range) != null;
```
@pimterry ?